tree/30_generic_methods/_service_check_running_smf.cf

```diff
@@ -36,14 +36,19 @@ bundle agent _service_check_running_smf(service_name)
       "class_prefix"        string => canonify(join("_", "promisers"));
       "args"                 slist => { "${service_name}" };
 
-  methods:
 
+  classes:
+      "report_mode"    expression => "${report_data.should_report}";
+
+  methods:
       "disable_reporting"     usebundle => disable_reporting;
 
+
       "check_if_service_is_running" usebundle => service_check_running("${service_name}");
 
-      "enable_reporting" usebundle => enable_reporting;
 
+      "reenable_reporting"    usebundle => enable_reporting,
+                             ifvarclass => "report_mode";
```

Comment:
You should use if instead of ifvarclass

Reply:
More details here https://github.com/Normation/ncf/pull/720

Reply:
@peckpeck You should read that: https://en.wikipedia.org/wiki/Broken_windows_theory